frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -1902,7 +1912,7 @@ void ProgramStructure::createControls() {
     std::vector<cstring> knownControls;
 
     for (auto it : controls)
-        knownControls.push_back(it.second);
+        knownControls.push_back(it.first->name);
```

Comment:
`ct` in line 1924 can also end up null and crash the program. Can you please add a null check there as well?